internal/tasks/schedulemanager_test.go

```diff
@@ -0,0 +1,81 @@
+package tasks
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+
+	mocks2 "github.com/razorpay/metro/internal/brokerstore/mocks"
+	"github.com/razorpay/metro/internal/common"
+	"github.com/razorpay/metro/pkg/registry/mocks"
+)
+
+func TestScheduleManager_Start(t *testing.T) {
```

Comment:
add UTs for `subscriptionmanager.go`? Do agree with its complexity. Can take it later if needed.

Reply:
Writing in a separate branch :)